recipes-bsp/rpi-eeprom/rpi-eeprom_git.bb

```diff
@@ -0,0 +1,66 @@
+SUMMARY = "Installation scripts and binaries for the Raspberry Pi 4 EEPROM"
+DESCRIPTION = "This repository contains the rpi4 bootloader and scripts \
+for updating it in the spi eeprom"
+LICENSE = "BSD-3-Clause & Broadcom-RPi"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=f546ed4f47e9d4c1fe954ecc9d3ef4f3"
+
+SRC_URI = " \
+    git://github.com/raspberrypi/rpi-eeprom.git;protocol=https;branch=master \
+"
+
+SRCREV = "6e79e995bbc75c5fdd5305bd7fe029758cfade2f"
+PV = "v2022.12.07-138a1"
+
+S = "${WORKDIR}/git"
+
+RDEPENDS:${PN} += " \
+    coreutils \
+    python3 \
+    python3-pycryptodomex \
+    openssl \
+    xxd \
+    pciutils \
+    userland \
```

Comment:
What's up with this dependency?

Reply:
Assuming you mean userland, the userland package provides the vcgencmd command which rpi-eeprom-config and rpi-eeprom-update can use to query the video core for the current bootloader configuration and version. I say "can" because the scripts do appear to support alternative methods of finding this information, but when executing the scripts they end up falling back to vcgencmd.

For a single example, see this hunk of code from rpi-eeprom-config.
```python
def read_current_config():
    """
    Reads the configuration used by the current bootloader.
    """
    fw_base = "/sys/firmware/devicetree/base/"
    nvmem_base = "/sys/bus/nvmem/devices/"

    if os.path.exists(fw_base + "/aliases/blconfig"):
        with open(fw_base + "/aliases/blconfig", "rb") as f:
            nvmem_ofnode_path = fw_base + f.read().decode('utf-8')
            for d in os.listdir(nvmem_base):
                if os.path.realpath(nvmem_base + d + "/of_node") in os.path.normpath(nvmem_ofnode_path):
                    return (open(nvmem_base + d + "/nvmem", "rb").read().decode('utf-8'), "blconfig device")

    return (shell_cmd(['vcgencmd', 'bootloader_config']), "vcgencmd bootloader_config")
```

In a default build using langdale, the nodes referenced in /sys/firmware/devicetree/base/ do exist, however /sys/bus/nvmem/devices/ is empty. For clarity, I've not done any serious investigation here since vcgencmd enables the scripts to work as-is.

Reply:
(besides this, everything else looks fine to me now)

Reply:
Your fix was merged. So that looks good. For the kernel config, I'm happy to start pulling in CONFIG_NVMEM_RMEM unconditionally for the relevant machines. So I'm up for a new cfg.

Reply:
OK, I've respun the branch with the following changes:

* Rebased branch onto latest master (tested with mickledore due to master changes).
* Updated rpi-eeprom busybox patch to say that it's a backport (merged but not yet part of a tagged release).
* Added config fragment for built in nvmem-rmem on raspberrypi4 machines.
* Updated rpi-eeprom commit message to explain the nvmem-rmem dependency.
* Tested on 5.15 kernel for raspberrypi4 and raspberrypi4-64.

Reply:
Looks good.